appinventor/components/src/com/google/appinventor/components/runtime/Form.java

```diff
@@ -311,6 +312,17 @@ public void onCreate(Bundle icicle) {
     // Called when the activity is first created
     super.onCreate(icicle);
 
+    // This version is for production apps. See {@link ReplForm#onCreate} for the REPL version,
+    // which overrides this method.
+    Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
+      @Override
+      public void uncaughtException(@NonNull Thread t, @NonNull Throwable e) {
+        Log.e(LOG_TAG, "Uncaught Exception", e);
+        ErrorOccurred(Form.this, "<unknown>",
```

Comment:
Doesn't this need to run on the UI thread?

Reply:
That's a good point. The documentation is unclear where the handler is run so we should play it safe.